### PR TITLE
Modal Welcome and Special Banner Breakout

### PR DESCRIPTION
**Pull request by srkirkland**
@a6wu @c3bryant @jpknoll  I really like this one, I broke out the welcome modal and "welcome banner" into their own components.  The 'WelcomeModal' now talks to realm directly and makes all the decisions about when it should show and writes to realm when it is dismissed.  It then shows a reusable 'InfoModal', which you could then use anywhere else in your app. Nice little self-contained widget.

The welcome banner I abstracted into a generic 'BannerView' which is just a nice touchable image which links to a website.  Then the new 'TopBannerView' is used to decide if the top banner should be show at all.  Currently it is using the same date/time logic to decide if it should show the welcome banner, but this would be a great place to have the top banner get its content from an API or static page so it could change based on the most important campus thing at the moment.

Note: Everything was running well but I couldn't test the last of the changes because it looks like the new mapping library is causing build error -- i get these even on your latest dev branch so it isn't related to this PR (error is in compiling /AIRMapPolygon.o).

### CODE REVIEW

File: app/views/banner/BannerView.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+} from 'react-native';
+
+var WebWrapper = require('../WebWrapper');
+var css = require('../../styles/css');
+
+// Card for displaying a large banner on the top of the screen
+export default class BannerView extends React.Component {
+  gotoWebView = () => {
+    // launch website
+    this.props.navigator.push({ id: 'WebWrapper', component: WebWrapper, title: this.props.site.title, webViewURL: this.props.site.url });
```
Comment (jpknoll) [inline app/views/banner/BannerView.js]: @srkirkland  Given the complexity of the props, do you want to add PropTypes to the BannerView class?

File: app/views/banner/TopBannerView.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+var general = 			  require('../../util/general');
+var AppSettings = 		require('../../AppSettings');
+
+// Manages the top 'hero' banner & decides what if anything to show
+// Currently decides based on static dates but an API would be great for breaking info
+export default class TopBannerView extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
```
Comment (jpknoll) [inline app/views/banner/TopBannerView.js]: @srkirkland I don't understand why this data is in state? Is this for future functionality (like an api call)?

File: app/views/banner/BannerView.js (hunk 1 above)
Comment (srkirkland) [inline app/views/banner/BannerView.js]: @jpknoll ooh i really do, just wasn't sure if we were ready for that yet.

**Comment by jpknoll** [comment]
@srkirkland Glad to see the realm pulled into it's own file :-D
:+1:

File: app/views/banner/TopBannerView.js (hunk 2 above)
Comment (srkirkland) [inline app/views/banner/TopBannerView.js]: @jpknoll yea that was my thought.  Eventually I'd like this component to get its data externally and then load the results into state.

**Pull request merged**